libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2727,28 +2727,41 @@ void IRGeneratorForStatements::assignInternalFunctionIDIfNotCalledDirectly(
 	m_context.addToInternalDispatch(_referencedFunction);
 }
 
-IRVariable IRGeneratorForStatements::convert(IRVariable const& _from, Type const& _to, bool _forceCleanup)
+IRVariable IRGeneratorForStatements::convert(IRVariable const& _from, Type const& _to)
 {
-	if (_from.type() == _to && !_forceCleanup)
+	if (_from.type() == _to)
 		return _from;
 	else
 	{
 		IRVariable converted(m_context.newYulVariable(), _to);
+		bool _forceCleanup = true;
 		define(converted, _from, _forceCleanup);
 		return converted;
 	}
 }
 
-std::string IRGeneratorForStatements::expressionAsType(Expression const& _expression, Type const& _to, bool _forceCleanup)
+IRVariable IRGeneratorForStatements::convertAndCleanup(IRVariable const& _from, Type const& _to)
+{
+	IRVariable converted(m_context.newYulVariable(), _to);
+	bool _forceCleanup = true;
+	define(converted, _from, _forceCleanup);
+	return converted;
+}
+
+std::string IRGeneratorForStatements::expressionAsType(Expression const& _expression, Type const& _to)
 {
 	IRVariable from(_expression);
 	if (from.type() == _to)
-	{
-		if (_forceCleanup)
-			return m_utils.cleanupFunction(_to) + "(" + from.commaSeparatedList() + ")";
-		else
-			return from.commaSeparatedList();
-	}
+		return from.commaSeparatedList();
+	else
+		return m_utils.conversionFunction(from.type(), _to) + "(" + from.commaSeparatedList() + ")";
+}
+
+std::string IRGeneratorForStatements::expressionAsCleanedType(Expression const& _expression, Type const& _to)
+{
+	IRVariable from(_expression);
+	if (from.type() == _to)
+		return m_utils.cleanupFunction(_to) + "(" + from.commaSeparatedList() + ")";
 	else
 		return m_utils.conversionFunction(from.type(), _to) + "(" + from.commaSeparatedList() + ")";
```

Comment:
It would be nice if we could re-use `expressionAsType` in this function.
Maybe
```suggestion
	IRVariable from(_expression);
	if (from.type() == _to)
		return m_utils.cleanupFunction(_to) + "(" + expressionAsType(_expression, _to) + ")";
	else
		return expressionAsType(_expression, _to) ;
```